common/meta/grpc/src/grpc_client.rs

```diff
@@ -25,72 +29,116 @@ use common_meta_types::protobuf::meta_client::MetaClient;
 
 impl MetaGrpcClient {
     pub async fn try_new(conf: &MetaGrpcClientConf) -> Result<MetaGrpcClient> {
-        Self::with_tls_conf(
-            &conf.meta_service_config.address,
-            &conf.meta_service_config.username,
-            &conf.meta_service_config.password,
-            Some(Duration::from_secs(conf.client_timeout_in_second)),
-            conf.meta_service_config.tls_conf.clone(),
-        )
-        .await
-    }
-
-    #[tracing::instrument(level = "debug", skip(password))]
-    pub async fn try_create(addr: &str, username: &str, password: &str) -> Result<Self> {
-        Self::with_tls_conf(addr, username, password, None, None).await
+        let mgr = MetaChannelManager {
+            timeout: Some(Duration::from_secs(conf.client_timeout_in_second)),
+            conf: conf.meta_service_config.tls_conf.clone(),
+        };
+        Ok(Self {
+            conn_pool: Pool::new(mgr, Duration::from_millis(50)),
+            addr: conf.meta_service_config.address.to_string(),
+            username: conf.meta_service_config.username.to_string(),
+            password: conf.meta_service_config.password.to_string(),
+            token: Arc::new(RwLock::new(None)),
+        })
     }
 
     #[tracing::instrument(level = "debug", skip(password))]
-    pub async fn with_tls_conf(
+    pub async fn try_create(
         addr: &str,
         username: &str,
         password: &str,
         timeout: Option<Duration>,
         conf: Option<RpcClientTlsConfig>,
     ) -> Result<Self> {
-        let res = ConnectionFactory::create_rpc_channel(addr, timeout, conf);
+        let mgr = MetaChannelManager { timeout, conf };
 
-        tracing::debug!("connecting to {}, res: {:?}", addr, res);
+        Ok(Self {
+            conn_pool: Pool::new(mgr, Duration::from_millis(50)),
+            addr: addr.to_string(),
+            username: username.to_string(),
+            password: password.to_string(),
+            token: Arc::new(RwLock::new(None)),
+        })
+    }
 
-        let channel = res?;
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub async fn make_client(
+        &self,
+    ) -> Result<MetaClient<InterceptedService<Channel, AuthInterceptor>>> {
+        let channel = self.conn_pool.get(&self.addr).await?;
+        tracing::debug!("connecting to {}, channel: {:?}", &self.addr, channel);
 
         let mut client = MetaClient::new(channel.clone());
-        let token = MetaGrpcClient::handshake(&mut client, username, password).await?;
-
-        let client = {
-            let token = token.clone();
-            MetaClient::with_interceptor(channel, AuthInterceptor { token })
+        let mut t = self.token.write().await;
+        let token = match t.clone() {
+            Some(t) => t,
+            None => MetaGrpcClient::handshake(&mut client, &self.username, &self.password).await?,
         };
 
-        let rx = Self { token, client };
-        Ok(rx)
+        *t = Some(token.clone());
```

Comment:
`*t = Some(token.clone());` 
This line can be placed inside the `None` branch of the above `match`?